### PR TITLE
add config to push log

### PR DESCRIPTION
**Pull request by ramaraochavali**
Slightly alternate solution to https://github.com/istio/istio/pull/33576

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu see if this acceptable?

File: pilot/pkg/xds/discovery.go [hunk 1]
```diff
@@ -455,6 +455,18 @@ func debounce(ch chan *model.PushRequest, stopCh <-chan struct{}, opts debounceO
...
 }
 
+func configsUpdated(req *model.PushRequest) string {
+	configs := ""
+	for key := range req.ConfigsUpdated {
+		configs += key.String()
+		break
+	}
+	if len(req.ConfigsUpdated) > 1 {
+		configs += " ..."
```
Comment (howardjohn) [inline pilot/pkg/xds/discovery.go:465]: nit: maybe say how many more?

**Review by howardjohn — COMMENTED** [review]
lgtm

File: pilot/pkg/xds/discovery.go [hunk 2]
```diff
@@ -407,8 +407,8 @@ func debounce(ch chan *model.PushRequest, stopCh <-chan struct{}, opts debounceO
 		if eventDelay >= opts.debounceMax || quietTime >= opts.debounceAfter {
 			if req != nil {
 				pushCounter++
-				log.Infof("Push debounce stable[%d] %d: %v since last change, %v since last push, full=%v",
-					pushCounter, debouncedEvents,
+				log.Infof("Push debounce stable[%d] %d for config %s: %v since last change, %v since last push, full=%v",
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/discovery.go:410]: It looks weired when `ConfigsUpdated == nil`

File: pilot/pkg/model/config.go [hunk 3]
```diff
@@ -59,6 +59,10 @@ func (key ConfigKey) HashCode() uint64 {
 	return binary.BigEndian.Uint64(sum)
 }
 
+func (key ConfigKey) String() string {
+	return key.Kind.String() + "/" + key.Name + "." + key.Namespace
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/config.go:63]: key.Kind.String() + "/" +  key.Namespace + "/" + key.Name

File: pilot/pkg/xds/discovery.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/xds/discovery.go:465]: > maybe say how many more?

Thats a great suggestion

**Comment by ramaraochavali** [comment]
@hzxuzhonghu addressed comments. PTAL

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**